cookbooks/aws-parallelcluster-scheduler-plugin/files/default/compute_fleet_status/update-compute-fleet-status.py

```diff
@@ -0,0 +1,98 @@
+import argparse
+import logging
+import sys
+import time
+from datetime import datetime, timezone
+
+import boto3
+from boto3.dynamodb.conditions import Attr
+
+DB_KEY = "COMPUTE_FLEET"
+DB_DATA = "Data"
+
+COMPUTE_FLEET_STATUS_ATTRIBUTE = "status"
+COMPUTE_FLEET_LAST_UPDATED_TIME_ATTRIBUTE = "lastStatusUpdatedTime"
+
+log = logging.getLogger()
+
+
+def update_item(table, status, current_status):
+    table.update_item(
+        Key={"Id": DB_KEY},
+        UpdateExpression="set #dt.#st=:s, #dt.#lut=:t",
+        ExpressionAttributeNames={
+            "#dt": DB_DATA,
+            "#st": COMPUTE_FLEET_STATUS_ATTRIBUTE,
+            "#lut": COMPUTE_FLEET_LAST_UPDATED_TIME_ATTRIBUTE,
+        },
+        ExpressionAttributeValues={
+            ":s": str(status),
+            ":t": str(datetime.now(tz=timezone.utc)),
+        },
+        ConditionExpression=Attr(f"{DB_DATA}.{COMPUTE_FLEET_STATUS_ATTRIBUTE}").eq(str(current_status)),
+    )
+
+
+def update_status_with_last_updated_time(table_name, region, status):
+    """Get compute fleet status and the last compute fleet status updated time."""
+    try:
+        table = boto3.resource("dynamodb", region_name=region).Table(table_name)
+        current_status = get_status(table)
+
+        if current_status == "UNKNOWN":
+            raise Exception("Could not retrieve compute fleet status.")
+        elif current_status == status:
+            return
+        update_item(table, status, current_status)
```

Comment:
let's enforce that PROTECTED status can be set only if current_status is RUNNING, this to mimic the current behavior, see https://github.com/aws/aws-parallelcluster-node/blob/develop/src/slurm_plugin/clustermgtd.py#L439-L443

Reply:
Added